target_chains/sui/contracts/sources/pyth.move

```diff
@@ -113,38 +162,85 @@ module pyth::pyth {
-                // registered with the Pyth State object.
-                if (!state::price_feed_object_exists(
-                        pyth_state,
-                            price_feed::get_price_identifier(
-                                price_info::get_price_feed(&cur_price_info)
-                            )
+            // Create and share new price info objects, if not already exists.
+            create_and_share_price_feeds_using_verified_price_infos(&latest_only, pyth_state, price_infos, ctx);
+        };
+        vector::destroy_empty(verified_vaas);
+    }
+
+    // create_and_share_price_feeds_using_verified_price_infos is a private function used by
+    // 1) create_price_feeds_using_batch_attestation
+    // 2) create_price_feeds_using_accumulator
+    // to create new price feeds for symbols.
+    fun create_and_share_price_feeds_using_verified_price_infos(latest_only: &LatestOnly, pyth_state: &mut PythState, price_infos: vector<PriceInfo>, ctx: &mut TxContext){
+        while (!vector::is_empty(&price_infos)){
+            let cur_price_info = vector::pop_back(&mut price_infos);
+
+            // Only create new Sui PriceInfoObject if not already
+            // registered with the Pyth State object.
+            if (!state::price_feed_object_exists(
+                    pyth_state,
+                        price_feed::get_price_identifier(
+                            price_info::get_price_feed(&cur_price_info)
                         )
-                ){
-                    // Create and share newly created Sui PriceInfoObject containing a price feed,
-                    // and then register a copy of its ID with State.
-                    let new_price_info_object = price_info::new_price_info_object(cur_price_info, ctx);
-                    let price_identifier = price_info::get_price_identifier(&cur_price_info);
-                    let id = price_info::uid_to_inner(&new_price_info_object);
+                    )
+            ){
+                // Create and share newly created Sui PriceInfoObject containing a price feed,
+                // and then register a copy of its ID with State.
+                let new_price_info_object = price_info::new_price_info_object(cur_price_info, ctx);
+                let price_identifier = price_info::get_price_identifier(&cur_price_info);
+                let id = price_info::uid_to_inner(&new_price_info_object);
 
-                    state::register_price_info_object(&latest_only, pyth_state, price_identifier, id);
+                state::register_price_info_object(latest_only, pyth_state, price_identifier, id);
 
-                    transfer::public_share_object(new_price_info_object);
-                }
+                transfer::public_share_object(new_price_info_object);
             }
+        }
+    }
+
+
+    // verified_vaa is the verified version of the VAA encoded within the accumulator_message
+    public fun create_authenticated_price_infos_using_accumulator(
```

Comment:
now that users can customize which price feeds are in the accumulator message, we could change the contract so that it always updates the on-chain state for *all* of the feeds in the message. This function could potentially do that work itself. I think that would simplify the interface for our users, as they wouldn't have to worry about `AuthenticatedVector<PriceInfo>`.

No need to do this change in this PR though.

Reply:
to the best of my knowledge the reason that it's not happening in the first place is a limitation of Sui to pass vector of resources in one call. Sui was supposed to fix it but not sure whether they did that.

Reply:
don't think it's possible atm